server/server.go

```diff
@@ -87,6 +95,20 @@ func (s *Server) RunContext(ctx context.Context) {
 	s.shutdown(context.Background())
 }
 
+func (s *Server) runMetricsServer() {
+	metricsAddr := s.Address + ":" + strconv.Itoa(s.MetricsPort)
```

Comment:
I'm unsure if using the same address here is the right thing.
I'm planning to use this in Kubernetes, so will control access externally to the process, meaning that I'm fine with using `0.0.0.0` and a different port number to keep `/metrics` internal only.
Thoughts?

Reply:
Wonder if it is easier to expose `Addr` config instead, and use Prometheus as the naming convention instead e.g. `WithPrometheusAddr`, `runPrometheusServer`

Reply:
Actually I am thinking perhaps a Prometheus metrics package that pulls metics from Go runtime and libvips (and server package should not meant to do that). Lets see what you think, maybe I can also make changes on top of your commits.

Reply:
Yeah, I can do a `metrics` package.

Reply:
@cshum ok, take a look.
I do like this better! :+1: